app/models/contestant.rb

```diff
@@ -1,3 +1,10 @@
 class Contestant < ApplicationRecord
   validates_presence_of :name, :age, :hometown, :years_of_experience
+
+  has_many :contestant_projects
+  has_many :projects, through: :contestant_projects, source: :project
```

Comment:
So, the `source: ` thing is not required - I think we talked about this maybe, a few folks were having this issue too - I think it's possible both sides of your many-to-many did not have the relationships & associations fleshed out first, and so Rails did not know where to go to find the contestant_projects association, which is why it insisted on you adding this part in. I would try to remove the `, source: :project` part from this line and see if it still works.